                                                                                                                                                                                                                                                               
Delivered-To: jwboyer@gmail.com
Received: by 10.76.27.197 with SMTP id v5csp113980oag;
        Mon, 27 Jan 2014 05:43:50 -0800 (PST)
X-Received: by 10.68.130.130 with SMTP id oe2mr3424108pbb.135.1390830230109;
        Mon, 27 Jan 2014 05:43:50 -0800 (PST)
Return-Path: <linux-kernel-owner@vger.kernel.org>
Received: from vger.kernel.org (vger.kernel.org. [209.132.180.67])
        by mx.google.com with ESMTP id rx8si11409416pac.134.2014.01.27.05.43.10
        for <multiple recipients>;
        Mon, 27 Jan 2014 05:43:50 -0800 (PST)
Received-SPF: pass (google.com: best guess record for domain of linux-kernel-owner@vger.kernel.org designates 209.132.180.67 as permitted sender) client-ip=209.132.180.67;
Authentication-Results: mx.google.com;
       spf=pass (google.com: best guess record for domain of linux-kernel-owner@vger.kernel.org designates 209.132.180.67 as permitted sender) smtp.mail=linux-kernel-owner@vger.kernel.org
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
	id S1753622AbaA0Njl (ORCPT <rfc822;padmanabhlinux@gmail.com>
	+ 99 others); Mon, 27 Jan 2014 08:39:41 -0500
Received: from mx1.redhat.com ([209.132.183.28]:64570 "EHLO mx1.redhat.com"
	rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
	id S1753380AbaA0Njl (ORCPT <rfc822;linux-kernel@vger.kernel.org>);
	Mon, 27 Jan 2014 08:39:41 -0500
Received: from int-mx09.intmail.prod.int.phx2.redhat.com (int-mx09.intmail.prod.int.phx2.redhat.com [10.5.11.22])
	by mx1.redhat.com (8.14.4/8.14.4) with ESMTP id s0RDdFjh026600
	(version=TLSv1/SSLv3 cipher=DHE-RSA-AES256-SHA bits=256 verify=OK);
	Mon, 27 Jan 2014 08:39:15 -0500
Received: from localhost (vpn1-5-245.ams2.redhat.com [10.36.5.245])
	by int-mx09.intmail.prod.int.phx2.redhat.com (8.14.4/8.14.4) with ESMTP id s0RDdDfN025726;
	Mon, 27 Jan 2014 08:39:14 -0500
From:	Francesco Fusco <ffusco@redhat.com>
To:	acme@infradead.org
Cc:	linux-kernel@vger.kernel.org
Subject: [PATCH] tools: perf: util: fix include for non x86 architectures
Date:	Mon, 27 Jan 2014 14:39:13 +0100
Message-Id: <2cf8143aad65a6aa6fe30325ef8a65847141afa2.1390829373.git.ffusco@redhat.com>
In-Reply-To: <cover.1390829373.git.ffusco@redhat.com>
References: <cover.1390829373.git.ffusco@redhat.com>
X-Scanned-By: MIMEDefang 2.68 on 10.5.11.22
Sender:	linux-kernel-owner@vger.kernel.org
Precedence: bulk
List-ID: <linux-kernel.vger.kernel.org>
X-Mailing-List:	linux-kernel@vger.kernel.org

Commit 71ae8aac ("lib: introduce arch optimized hash library")
added an include to <linux/hash.h> for setting up an architecture
specific fast hash. Since perf includes directly the non-uapi
kernel header, it cannot find <asm/hash.h> on non-x86 and thus
prevents perf to be compiled on every architecture other than
x86. The problem is the inclusion of <asm/hash.h> in hash.h
that results in the following error originating from
util/evlist.c:

  fatal error: asm/hash.h: No such file or directory

This commit simply adds an empty <asm/hash.h> stub/file to fix
the compile issue on non-x86 architectures. As perf does not use
any of these new functions, it fixes the compilation and therefore
seems to be the most appropriate solution to go with.

Signed-off-by: Francesco Fusco <ffusco@redhat.com>
---
 tools/perf/util/include/asm/hash.h | 6 ++++++
 1 file changed, 6 insertions(+)
 create mode 100644 tools/perf/util/include/asm/hash.h

diff --git a/tools/perf/util/include/asm/hash.h b/tools/perf/util/include/asm/hash.h
new file mode 100644
index 0000000..d82b170b
--- /dev/null
+++ b/tools/perf/util/include/asm/hash.h
@@ -0,0 +1,6 @@
+#ifndef __ASM_GENERIC_HASH_H
+#define __ASM_GENERIC_HASH_H
+
+/* Stub */
+
+#endif /* __ASM_GENERIC_HASH_H */
-- 
1.8.3.1

--
To unsubscribe from this list: send the line "unsubscribe linux-kernel" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html
Please read the FAQ at  http://www.tux.org/lkml/
